### ISSUE TITLE
Search component is missing a visible label

### ISSUE THREAD

**Issue by ernon66** [Sage/carbon#7603]
### Description

I would expect to be able to add a visible label to the Search component

An input without a visible label can be an accessibility failure

### Reproduction

https://carbon.sage.com/?path=/docs/search--docs

### JIRA ticket numbers (Sage only)

SBS-145230

### Suggested solution

Ideally would like to have a `label` prop like Textbox that can also be inline

Otherwise just surfacing `aria-labelledby` would work but would be heavier to use

### Carbon version

158.0.0

### React version

v17

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.
